libcaf_net/caf/net/basp/application.hpp

```diff
@@ -136,7 +136,8 @@ class application {
     // nop
   }
 
-  static expected<buffer_type> serialize(actor_system& sys, const message& x);
+  static error_code<sec> serialize(actor_system& sys, const message& x,
+                                   std::vector<byte>& buf);
```

Comment:
```suggestion
                                   byte_buffer& buf);
```

Recent versions of CAF have a typedef for this: https://github.com/actor-framework/actor-framework/blob/master/libcaf_core/caf/byte_buffer.hpp. We should use it consistently.

Reply:
I'll try and replace all occurrences.